### DISCUSSION TITLE
Backport "Large Corkscrew" element from RCT3 into OpenRCT2

### DISCUSSION THREAD

**Discussion by Shima33** [OpenRCT2/OpenRCT2#15755] [Feature requests] [answered]
When I was young and didn't know not know how to manage my G-forces, it was always Corkscrews that ruined my designs. Now that I'm older, I still approach them with a great deal of apprehension, simply because of how many G's they put on the designs.

When they added the Large Corkscrew to RCT3, it felt like a blessing - finally I could have corkscrew elements in my coasters that didn't add stupid amounts of G's. Even today, I see players, even experienced ones, getting tripped up by just how many G's a Corkscrew entails - a Large Corkscrew, like the one in RCT3, would be a life saver for quite a lot of builders.

This might be something quite difficult to add because of the requirement of new sprites and all the rest of it, so I wouldn't be surprised if this was limited to a certain coaster type (Maybe just B&M's?) , but I think that the more coasters have this, the more flexibility they'll be for track designers.

**Comment by ocalhoun6** [comment]
Maybe a good candidate to start with would be the hybrid coaster that just got merged? (And then we could work on adding it to other track designs as time goes on. Once it's developed for one coaster type, adding it to others should just be a matter of adding the appropriate sprites.)

It's still pretty fresh, and the guy who made it is still around, so making one more type of track element for it might not be too hard.

Also, since it's an original coaster type, nobody will be mad at us for 'ruining' an existing coaster type by adding non-vanilla track pieces to it.

----------------------

And while we're dreaming up new track sections, I think a ['beyond vertical' drop](https://media.techeblog.com/images/takabisha-roller-coaster.jpg) is sorely missing from RCT2. There's no really good (easy) way to duplicate it. Though it probably wouldn't be a good fit for the hybrid coaster, since I don't think there are any real-world examples of hybrid coasters that use it.

But it could definitely be added as a new one-piece special element of a vertical drop coaster or maybe a twister.

**Reply by ThiaJay** [reply]
Beyond vertical drops are popular these days. Whilst we're at it, it would be nice to go from an inversion into upside-down track pieces for as long as we want rather than forcing a loop, corkscrew, barrel roll or twist. That would enable a lot of flexibility. Even better, being able to make our own loops by then going into vertical and then sloped pieces back to level.

**Comment by SpartanFrederic104** [comment]
All of these new track pieces you talk about and wishlist are best saved for when the new save format is released.

**Comment by Ryder17z** [comment]
> When they added the Large Corkscrew to RCT3, it felt like a blessing - finally I could have corkscrew elements in my coasters that didn't add stupid amounts of G's. Even today, I see players, even experienced ones, getting tripped up by just how many G's a Corkscrew entails - a Large Corkscrew, like the one in RCT3, would be a life saver for quite a lot of builders.

How do you feel about large loops?

**Comment by ghost** [comment]
> > When they added the Large Corkscrew to RCT3, it felt like a blessing - finally I could have corkscrew elements in my coasters that didn't add stupid amounts of G's. Even today, I see players, even experienced ones, getting tripped up by just how many G's a Corkscrew entails - a Large Corkscrew, like the one in RCT3, would be a life saver for quite a lot of builders.
> 
> How do you feel about large loops?

Aren't those already in the game?

**Comment by X123M3-256** [comment]
This requires the new save format (#10664). New track elements cannot be added right now because the game currently uses a byte to store the track piece ID, which means there's a limit of 255 pieces, all of which are used.

**Comment by X123M3-256** [comment]
I now have a working implementation of large corkscrews, based on the new save format branch. They're not a direct backport of the RCT3 version though, mine are slightly taller and the clearance boxes are different (honestly, unless we want to add the ability to import RCT3 save files someday, I see little reason to port the RCT3 version and would prefer to have the flexibility to design it from scratch).

https://i.imgur.com/KVSmarn.gif

It'll be some time yet before something like this can be merged, however. I now realize that this not only requires the new save format, but it would also require a new track design format, as TD6 files can't handle these new elements either.

**Comment by Gymnasiast** [comment, marked as answer]
@X123M3-256 ’s work has been merged in the meantime.
